File: templates/src/main/resources/templates/minimal/lang/java/src/main/java/{{packagePath}}/{{class}}.java
package {{package}};

import dev.ancaria.coderpack.api.Context;
import dev.ancaria.coderpack.api.SacredMod;

/** {{description}} */
public final class {{class}} implements SacredMod {

    @Override
    public void onLoad(Context context) {
        context.log("Loaded.");
    }
}

File: templates/src/main/resources/templates/minimal/lang/kotlin/src/main/kotlin/{{packagePath}}/{{class}}.kt
package {{package}}

import dev.ancaria.coderpack.api.Context
import dev.ancaria.coderpack.ktx.SacredMod

/** {{description}} */
class {{class}} : SacredMod() {

    /**
     * Runs once, with the loader's context as the receiver. The same object
     * stays available as `context` from anywhere else in the class.
     */
    override fun Context.load() {
        log("Loaded.")
    }
}
